tensorflow_datasets/core/load.py

```diff
@@ -179,6 +180,65 @@ def builder(
     return builder_cls(name)(**builder_kwargs)  # pytype: disable=not-instantiable
 
 
+def find_builder(name: str, *, data_dir: Optional[str] = None) -> Optional[str]:
+  """Returns the path to dataset files
+
+  Example:
+
+  ```py
+  path = find_builder_dir('ds/config:1.2.0', data_dir='~/tensorflow_datasets/')
+  asssert path == '~/tensorflow_datasets/ds/config/1.2.0'
+
+  # If the version is not specified, return the last available version
+  find_builder_dir('ds', data_dir='~/tensorflow_datasets/')
+  assert path == '~/tensorflow_datasets/my_dataset/config/1.2.0'
+  ```
+
+  """
+  ds_name, name_kwargs = _dataset_name_and_kwargs_from_name_str(name)
+
+  base_dir = os.path.join(data_dir or constants.DATA_DIR, ds_name)
+  config = name_kwargs.get("config", None)
+  version = name_kwargs.get("version", None)
+
+  if config and version:
+    builder_dir = os.path.join(base_dir, config, version)
+    if tf.io.gfile.exists(builder_dir):
+      return builder_dir
+
+  if version:
+    builder_dir = os.path.join(base_dir, version)
+    if tf.io.gfile.exists(builder_dir):
+      return builder_dir
+
+  config_version = _select_config_and_version(base_dir)
+  if config_version:
+    return os.path.join(base_dir, config_version)
+
+  version = _select_version(base_dir)
+  if version:
+    return os.path.join(base_dir, version)
+
+  return None
+
+def _select_config_and_version(data_dir: str) -> Optional[str]:
+  config = tf.io.gfile.listdir(data_dir)[0]  # TODO: Select default config
```

Comment:
Done for now.

In the future, we will change this right? (as we are reading the dataset code)

Reply:
Yes, I don't know what should be the good long term solution. Maybe adding a `dataset/info.json` file in the top folder ? Or somehow add some local clue to set a folder as the default one.

Reply:
Or maybe we could just warn the users to specify the config name. The default behaviour would be to select the first config in the sorted order

Reply:
Select the first config in sorted order is very dangerous, for instance, if one user generate `wikipedia/en`, and another user generate both `wikipedia/en` and `wikipedia/de`, using `tfds.load('wikipedia')` will load a different dataset depending for each users.

I think it's best to be explicit in this case:
* If the code is reachable, extract the default builderConfig
* Otherwise, raise an error -> Please specify the config name

Reply:
Okay, got it.
Made the changes